modules/component-store/src/component-store.ts

```diff
@@ -128,23 +132,26 @@ export class ComponentStore<T extends object> implements OnDestroy {
               : // If state was not initialized, we'll throw an error.
                 throwError(() => new Error(this.notInitializedErrorMessage))
           ),
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          map(([value, currentState]) => updaterFn(currentState, value!)),
+          tap((newState) => this.stateSubject$.next(newState)),
+          catchError((error: unknown) => {
+            if (isSyncUpdate) {
+              syncError = error;
+              return EMPTY;
+            }
+
+            return throwError(() => error);
+          }),
           takeUntil(this.destroy$)
         )
-        .subscribe({
-          next: ([value, currentState]) => {
-            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-            this.stateSubject$.next(updaterFn(currentState, value!));
-          },
-          error: (error: Error) => {
-            initializationError = error;
-            this.stateSubject$.error(error);
-          },
-        });
+        .subscribe();
 
-      if (initializationError) {
-        // prettier-ignore
-        throw /** @type {!Error} */ (initializationError);
+      if (syncError) {
+        throw syncError;
```

Comment:
` /** @type {!Error} */` was needed for Google's Closure compiler, and `\\prettier-ignore` was needed so it doesn't break the lines.

Don't know if `unknown` changes that - we'd need to validate with our friends at Google.

Reply:
Is there a way that we can test this ourselves?
Or should we add a comment that this was done for the Closure compiler?
The reason why I ask this is that this would have been overlooked if you didn't review it.

Reply:
Let's check with them. If it's still needed, I'll add a comment.

Reply:
Googlers confirmed that the type comment is not needed anymore. The PR is ready for re-review 👀 cc @alex-okrushko